### PR TITLE
[SPARK-10617] [SQL] Fixed AddMonths leap year calculations

### PR DESCRIPTION
**Pull request by alexrovner**

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala [hunk 1]
```diff
@@ -295,6 +295,8 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
   test("add_months") {
     checkEvaluation(AddMonths(Literal(Date.valueOf("2015-01-30")), Literal(1)),
       DateTimeUtils.fromJavaDate(Date.valueOf("2015-02-28")))
+    checkEvaluation(AddMonths(Literal(Date.valueOf("2015-02-28")), Literal(12)),
+      DateTimeUtils.fromJavaDate(Date.valueOf("2016-02-28")))
```
Comment (davies) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala]: This is not the expected answer, Hive/Oracle will return 2016/02/29

**Comment by andrewor14** [comment]
ok to test

**Comment by andrewor14** [comment]
@alexrovner actually this PR is opened against the wrong branch. Would you mind closing it and re-opening against the master branch?

**Comment by davies** [comment]
@alexrovner Could you close this PR since the JIRA had been closed.

**Pull request closed without merging**